### PR TITLE
Make Program['comments'] the source of truth

### PR DESCRIPTION
**Pull request by sebmck**
Right now we have three properties on all nodes: `innerComments`, `trailingComments`, and `leadingComments`. We also have a property on `Program` called `comments` that contains all the comments in the original source document. Comments can reside on multiple nodes, so they are duplicated. This makes sense since it's not clear how a comment relates to a particular node so it's important to have them visible in multiple places.

However this makes it so you can't rely on `Program['comments']` being up to date, and when you update one of the comment nodes, it might still be modified on another node, and if the order changes then your comment modification means nothing.

This PR changes all of those comment arrays to `Array<string>`. The `string` is a unique ID that refers to the comment in `Program['comments']`. We then have some bookkeeping that allows you to resolve these correctly, and gets updated.

Inside of a transform instead of doing `node.innerComments` you would now do `context.getComments(node.innerComments)`.

There is a hook exposed for comment insertion, so to insert a new comment you would do:

```
return {
  ...node,
  leadingComments: [path.callHook(commentInjectorVisitor, {type: 'CommentBlock', value: 'ok'})]
};
```

This call will add the comment to `Program`, and return an id.

### CODE REVIEW

File: packages/@romejs/js-compiler/api/analyzeDependencies.test.ts [hunk 1]
```diff
@@ -24,22 +24,6 @@ async function testAnalyzeDeps(input: string, sourceType: ConstSourceType) {
   });
 }
 
-test("discovers require('module') call", async (t) => {
```
Comment (diokey) [inline packages/@romejs/js-compiler/api/analyzeDependencies.test.ts:27]: why was this removed?

Comment (sebmck) [reply packages/@romejs/js-compiler/api/analyzeDependencies.test.ts:27]: Oops. `test.only` didn't seem to be working so I just deleted the other tests. Will add them back.
